rest-server/templates/src/main/scala/com/actionml/templates/cb/CBEngine.scala

```diff
@@ -68,7 +75,8 @@ class CBEngine() extends Engine() with JsonParser {
   }
 
   /** Triggers parse, validation, and persistence of event encoded in the json */
-  def input(json: String, trainNow: Boolean = true): Validated[ValidateError, Boolean] = {
+  // Slava this needs to stay "input" why did you change the API?
+  def inputInternal(json: String, trainNow: Boolean = true): Validated[ValidateError, Boolean] = {
```

Comment:
I just meant that `inputInternal` is not a part of public API. I am ok with removing this method but I do not know how to implement JSON mirroring on `Engine` level another way.